content/Sessions/Sessions.q.md

```diff
@@ -0,0 +1,924 @@
+
+ - It returns the `$_SESSION["firstName"]` session variable.
+
+ - It displays the `$_SESSION["firstName"]` session variable.
+
+ - It accesses the `$_SESSION["firstName"]` session variable.
+
+:::
+
+
+:::
+
+/// type=REPL, readonly=true
+
+```php
+<?php
+session_start();
+
+if (!empty($_POST)) {
+    if (isset($_POST['username']) && isset($_POST['password'])) {
+        echo "Hello, " . $_POST['username'] . '.';
+    }
+}
+?>
+
+<html>
+    <body>
+        <form action="" method="post">
+            <input type="text" name="username" placeholder="Username" required>
+            <input type="password" name="password" placeholder="Password" required>
+            <input type="submit" value="Submit">
+        </form>
+    </body>
+</html>
+```
+/// type=SS, choiceType=HTML, answer=[5]
+
+Execute the program. What is its output?
+
+ - <form action="" method="post">
+            <input type="submit" value="Submit">
+            <input type="password" name="password" placeholder="Password" required> 
+    </form>
+
+ - <form action="" method="post">
+            <input type="password" name="password" placeholder="Password" required>
+            <input type="submit" value="Submit">
+    </form>
+
+ - <form action="" method="post">
+            <input type="text" name="username" placeholder="Username" required>
+            <input type="submit" value="Submit">
+    </form>
+
+ - <form action="" method="post">
+            <input type="text" name="username" placeholder="Username" required>
+            <input type="password" name="password" placeholder="Password" required>
+    </form>
+
+ - <form action="" method="post">
+            <input type="text" name="username" placeholder="Username" required>
+            <input type="password" name="password" placeholder="Password" required>
+            <input type="submit" value="Submit">
+    </form>
+
+
+/// type=SS, answer=[5]
+
+In the browser window, click the `Submit` button. What is its result?
+
+ - It prints `Hello`.
+
+ - It produces an error.
+
+ - No output is displayed.
+
+ - It prints `Hello, user.`.
+
+ - It displays a message `This is a required field`.
+
+
+/// type=SS, answer=[5]
+
+In the browser window, replace `Username` with `Charles` and `Password` with `12345`. Click the `Submit` button. What is its output?
+
+ - It prints `Hello`.
+
+ - It produces an error.
+
+ - No output is displayed.
+
+ - It prints `Hello, user.`.
+
+ - It prints `Hello, Charles.`.
+
+
+/// type=MS, answer=[2,3,4]
+
+Which of the following are functions?
+
+ - `$_POST`
+
+ - `empty()`
+
+ - `isset()`
+
+ - `session_start()`
+
+ - `$_POST['password']`
+
+
+/// type=SS, answer=[5]
+
+In the `if` statement on line 5, what is `$_POST`?
+
+ - It is a keyword.
+
+ - It is a function.
+
+ - It is a construct.
+
+ - It is an operator.
+
+ - It is a superglobal variable.
+
+
+/// type=SS, answer=[5]
+
+In the `if` statement on line 5, what does `$_POST['username']` do?
```

Comment:
For the options here, is it okay to refer to `username` as a key?
Similar to this option on line 632?
> On line 2, the `name` key in the `$_SESSION` superglobal variable does not exist.

Or as a field

> It accesses the value of the `username` field from the form...

Reply:
Yes, this is referred to as `key` and not `field` since `$_POST[]` is treated as an array and what is specified inside the square brackets is referred to as the `key`.